webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/internal/EventLogAPI.scala

```diff
@@ -0,0 +1,161 @@
+      case Full(events) => Full((events.size, events.slice(start,start+nbelement)))
+      case eb: EmptyBox  => eb ?~! s"Error when trying fetch eventlogs from database for page ${(start/nbelement)+1}"
+    }
+  }
+
+  def requestDispatch: PartialFunction[Req, () => Box[LiftResponse]] = {
+    case Get(Nil, req) =>
+
+      val draw = req.params.get("draw") match {
+        case Some(value :: Nil) => Full(value.toInt)
+        case None => Failure("Missing 'draw' field from datatable's request")
+      }
+      val start = req.params.get("start") match {
+        case Some(value :: Nil) => Full(value)
+        case None => Failure("Missing 'start' field from datatable's request")
+      }
+      val length = req.params.get("length") match {
+        case Some(value :: Nil) => Full(value)
+        case None => Failure("Missing 'length' field from datatable's request")
+      }
+
+      val response = (draw, start, length) match {
+        case (Full(d), Full(s), Full(l)) =>
+          repos.getEventLogCount.toBox  match  {
+            case Full(totalRecord) =>
+              getEventLogBySlice(s.toInt, l.toInt, None,  None,  Some("creationdate DESC" )) match {
+                case Full((totalFiltered, events)) =>
+                  responseFormater(d, totalRecord,  totalFiltered, events)
+                case eb:  EmptyBox                 =>
+                  val fail = eb  ?~! "Failed to get eventlogs"
+                  responseFormater(d, totalRecord,  0, Vector.empty, Some(fail.messageChain))
+              }
+            case eb: EmptyBox      =>
+              val fail = eb ?~! "Failed to get event log's count"
+              responseFormater(d, 0,  0, Vector.empty, Some(fail.messageChain))
+          }
+
+        case (ebDraw: EmptyBox,_, _)      =>
+          val fail = ebDraw ?~! "Missing parameter in request"
+          responseFormater(0, 0, 0, Vector.empty, Some(fail.messageChain))
+
+        case (_,ebStart: EmptyBox, _)     =>
+          val fail = ebStart ?~! "Missing parameter in request"
+          responseFormater(0, 0, 0, Vector.empty, Some(fail.messageChain))
+
+        case (_,_, ebLength: EmptyBox)    =>
+          val fail = ebLength ?~! "Missing parameter in request"
+          responseFormater(0, 0, 0, Vector.empty, Some(fail.messageChain))
+      }
+      JsonResponse(response, Nil, Nil, 200)
+
+    case Get(id :: "details" :: Nil, _) =>
+      repos.getEventLogByCriteria(Some(s"id = $id")).toBox match {
+        case Full(e) =>
+          e.headOption match {
+            case Some(eventLog) =>
+              val crid = eventLog.id.flatMap(repos.getEventLogWithChangeRequest(_).toBox match {
+                case Full(Some((_, crId))) => crId
+                case _                     => None
+              })
+              val htmlDetails = eventLogDetail.displayDetails(eventLog, crid)
+              toJsonResponse(None, "content" -> htmlDetails.toString())("eventdetails", prettify = false)
```

Comment:
The toString here breaks JS script that used to be returned along with the HTML to diplay diff